feat(agent): implement agentic Text2GQL system with multi-stage workflow

SeasonPilot opened the pull request:
## Summary

https://github.com/TuGraph-family/chat2graph/issues/231

Implement comprehensive agentic text-to-graph query language system with context awareness, multi-stage validation, and self-optimization capabilities.

Core Components:
- DAL Layer: Query session, history, pattern, and feedback management (8 files)
- Service Layer: Context tools and query context service (2 files)
- Operators: 6 specialized operators for query lifecycle management
  * Query intention analysis and validation
  * Complexity analysis with index recommendations
  * Path pattern recognition (multi-hop, temporal, spatial)
  * Comprehensive query validation (schema, semantic, performance, security)
  * Context enhancement with user preference learning
  * Query design with optimization strategies
- Resources: Multi-hop reasoning, query planning, and validation (3 files)
- Workflow: Complete agentic text2gql configuration with 7-stage pipeline
- Testing: Comprehensive test suite (22 files: unit + integration)
  * Unit tests for DAOs and services
  * Integration tests for operator collaboration and workflows

Key Features:
- Multi-stage workflow: understand → analyze → enhance → design → validate → execute
- Context-aware query optimization with historical pattern learning
- Real-time validation (schema, semantic, performance, security)
- Support for complex patterns: multi-hop, temporal, spatial queries
- Self-validating agentic workflow with feedback loops

<!--
Before opening your pull request, have a quick look at our contribution guidelines:
https://github.com/TuGraph-family/community/blob/master/docs/CONTRIBUTING.md
-->

## Title
<!--
And make sure that the title of your pull request follows the following format:
`<type>(<scope>): <subject>`

`<type>` is the type of your pull request.
`<scope>` is optional (including `()`) when you choose `none`.
`<subject>` is a concise sentence in lowercase.
-->

**Type**
<!-- What is the type of your pull request? Open the item by `[x]` way. -->

- [x] `feat`: (new feature)
- [ ] `fix`: (bug fix)
- [ ] `docs`: (doc update)
- [ ] `style`: (update format)
- [ ] `refactor`: (refactor code)
- [ ] `test`: (test code)
- [ ] `chore`: (other updates)

**Scope**
<!-- Which module does your pull request mainly modify? Select `none` when undetermined. -->

- [x] `app`: (**Application Layer**)
  - [ ] `web`: (web front-end module)
  - [ ] `server`: (web server module)
  - [x] `dal`: (data access layer)
  - [ ] `sdk`: (sdk module)
- [ ] `agent`: (**Agent Layer**)
  - [x] `reasoner`: (reasoner module)
  - [ ] `planner`: (planner module)
  - [ ] `workflow`: (workflow module)
  - [ ] `memory`: (memory module)
  - [ ] `knowledge`: (knowledge module)
  - [ ] `env`: (env module)
  - [ ] `toolkit`: (toolkit module)
- [ ] `system`: (**System Layer**)
  - [x] `plugin`: (plugin module)
  - [ ] `tracer`: (tracer module)
  - [ ] `resource`: (resource module)
- [ ] `none`: (N/A)

### Description
<!-- Provide the relevant issue number associated with your pull request if needed. -->

**Issue:** #

<!-- Provide more information about this pull request. -->

### Checklist

- [x] I have prepared the pull request title according to the requirements.
- [x] I have successfully run all unit tests and integration tests.
- [x] I have followed the code style guidelines of this project.
- [x] I have already rebased the latest `master` branch.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.

SeasonPilot committed:
feat(agent): implement agentic text-to-GQL system with multi-stage workflow

Implement comprehensive agentic text-to-graph query language system with
context awareness, multi-stage validation, and self-optimization capabilities.

Core Components:
- DAL Layer: Query session, history, pattern, and feedback management (8 files)
- Service Layer: Context tools and query context service (2 files)
- Operators: 6 specialized operators for query lifecycle management
  * Query intention analysis and validation
  * Complexity analysis with index recommendations
  * Path pattern recognition (multi-hop, temporal, spatial)
  * Comprehensive query validation (schema, semantic, performance, security)
  * Context enhancement with user preference learning
  * Query design with optimization strategies
- Resources: Multi-hop reasoning, query planning, and validation (3 files)
- Workflow: Complete agentic text2gql configuration with 7-stage pipeline
- Testing: Comprehensive test suite (22 files: unit + integration)
  * Unit tests for DAOs and services
  * Integration tests for operator collaboration and workflows

Key Features:
- Multi-stage workflow: understand → analyze → enhance → design → validate → execute
- Context-aware query optimization with historical pattern learning
- Real-time validation (schema, semantic, performance, security)
- Support for complex patterns: multi-hop, temporal, spatial queries
- Self-validating agentic workflow with feedback loops

SeasonPilot commented:
#231

qishipengqsp commented on test/unit/plugin/tugraph/__init__.py:
remove this

qishipengqsp reviewed (changes requested):
Remove the Chinese comments and file headers, aligning to the project

SeasonPilot committed:
refactor(text2gql): address PR #233 code review comments

- Move example file and workflow config to test/example/ directory
- Remove empty __init__.py files in test/unit/plugin/tugraph/
- Fix unused variables in example file (c2g removed, job -> _job)
- Remove unused imports (Dict, Any, Expert, AgentMessage, Chat2Graph)
- Add comprehensive error handling to Query Execution operator
  - Error categories: CONNECTION, TIMEOUT, SYNTAX, PERMISSION, RESOURCE
  - Retry strategy with exponential backoff for transient errors
  - Enhanced output schema with recovery suggestions

SeasonPilot commented on test/example/agentic_text2gql.yml:
```diff
@@ -0,0 +1,503 @@
+      - *query_design_action
+
+  # Operator 7: Query Execution
+  - &query_execution_operator
```
Added comprehensive error handling to Query Execution operator in commit 4391c50:

- **Error categories**: CONNECTION, TIMEOUT, SYNTAX, PERMISSION, RESOURCE
- **Retry strategy**: Exponential backoff (1s, 2s, 4s) for transient errors, max 3 retries
- **Enhanced output schema**: Includes error_handling block with recovery suggestions

Non-transient errors (syntax, permission) do not trigger retries.

qishipengqsp committed:
Merge branch 'master' into feat/agent-agentic-text2gql
